meson.build

```diff
@@ -14,6 +14,9 @@ if fcc.get_id() == 'gcc'
                            '-ffpe-trap=invalid,zero,overflow'],
                           language: 'fortran')
   endif
+  if get_option('buildtype') == 'release'
+    add_project_arguments('-ffp-contract=off'.split(),language: 'fortran')
+  endif
```

Comment:
I see from [gcc documentation](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html) that this option disables floating-point expression contraction.
- why do we want to use this option? Is it for performance or accuracy?
- why is it only enabled for buildtype == release?

Reply:
By using the "-ffp-contract=off" option, optimizations that alter the result of floating-point expressions, such as fused multiply-add (FMA), are omitted. This gives better reproducibility between different optimization levels. It is somewhat the same as the use of "-fp-model precise" for intel compilers. Some performance is lost, but errors discovered in optimized code can easier by debugged with no optimization and debug flags.